src/java/org/apache/cassandra/metrics/HintsServiceMetrics.java

```diff
@@ -51,6 +64,15 @@ public final class HintsServiceMetrics
                                                                                                .executor(ImmediateExecutor.INSTANCE)
                                                                                                .build(address -> Metrics.histogram(factory.createMetricName("Hint_delays-"+address.toString().replace(':', '.')), false));
 
+    private static class TotalHintsSizeGauge implements Gauge<Long>, Serializable
```

Comment:
I could not use method reference like

````
Metrics.gauge(factory.createMetricName("HintsFileSize"), HintsService.instance::getTotalHintsSizeOfNode);
````

because at the time of that static variable being initialized, HintsService.instance is `null` / is not initialized yet so this is the workaround.

Reply:
This comment should be a javadoc comment :-) I suppose